src/cli/mod.rs

```diff
@@ -0,0 +1,11 @@
+mod config;
+use crate::api_client;
+
+pub use config::{BeaconMethod, CliConfig, Namespace::Beacon};
+
+pub async fn run_cli(client: api_client::Client, args: CliConfig) {
+    match args.command {
+        Beacon(BeaconMethod::Genesis(genesis)) => genesis.execute(&client).await,
+        Beacon(BeaconMethod::Root(ref root_arg)) => root_arg.execute(&client).await,
+    }
```

Comment:
I believe if you move the refs to the signature, you can do this (which imo is a bit simpler code)

and also, if I was doing this and knew there would be a lot of copy-pasta to potentially do, I'd structure it so that the names are generic enough so that they make sense on each individual line but can still be easily copied across lines -- so I'd just call the inner thing an `arg`, as ultimately we just do the dispatch to figure out which namespace/method we want and then call the rest of the logic w/ whatever arguments we discovered from the cli parse

```suggestion
    match args.command {
        Beacon(BeaconMethod::Genesis(arg)) => arg.execute(client).await,
        Beacon(BeaconMethod::Root(arg)) => arg.execute(client).await,
    }
```